Spigot-Server-Patches/0551-Use-block-distance-in-portal-search-radius.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Patrick Hemmer <patrick.hemmer@gmail.com>
+Date: Fri, 7 Aug 2020 23:22:43 -0400
+Subject: [PATCH] Use block distance in portal search radius
+
+Previously when searching for a portal, the server would use the search radius to determine which chunks it needs to search, and then would consider any portal within those chunks as a match, even if it is outside the search radius.
+This changes it so that the portal list is filtered to only include portals within the configured search radius (ignoring the Y coordinate).
+
+diff --git a/src/main/java/net/minecraft/server/PortalTravelAgent.java b/src/main/java/net/minecraft/server/PortalTravelAgent.java
+index f6ec165f9c7d83698100c85b1d4bf6b4cea0f458..29ea04452a12c8a5d460e6c6a2fdab297448e5aa 100644
+--- a/src/main/java/net/minecraft/server/PortalTravelAgent.java
++++ b/src/main/java/net/minecraft/server/PortalTravelAgent.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import java.util.Collections;
+ import java.util.Comparator;
+ import java.util.List;
+ import java.util.Optional;
+@@ -57,11 +58,18 @@ public class PortalTravelAgent {
+         List<VillagePlaceRecord> list = (List) villageplace.b((villageplacetype) -> {
+             return villageplacetype == VillagePlaceType.v;
+         }, blockposition, searchRadius, VillagePlace.Occupancy.ANY).collect(Collectors.toList()); // CraftBukkit - searchRadius
+-        Optional<VillagePlaceRecord> optional = list.stream().min(Comparator.<VillagePlaceRecord>comparingDouble((villageplacerecord) -> { // CraftBukkit - decompile error
+-            return villageplacerecord.f().j(blockposition);
+-        }).thenComparingInt((villageplacerecord) -> {
+-            return villageplacerecord.f().getY();
+-        }));
++        // Paper start - filter by searchRadius
++        list.removeIf(villagePlaceRecord -> {
++            BlockPosition portalPosition = villagePlaceRecord.f();
```

Comment:
This is a behavioral change vs Vanilla unless I'm missing something? So we need a config option for this.

However, I'd put it with the existing search radius configs block.

If your ok with not having authorship on the patch file itself, could merge with the previous.
But otherwise ok with a separate patch, just bundle the config next to the others.

Reply:
Done.